src/components/datagrid.js

```diff
@@ -25,9 +25,9 @@ module.exports = function(app) {
             view += '<tr>';
             formioUtils.eachComponent(component.components, function(component) {
               // Don't render disabled fields, or fields with undefined data.
-              if (!component.tableView || row[component.key] === undefined) {
-                return;
-              }
+              // if (!component.tableView || row[component.key] === undefined) {
```

Comment:
Why did we comment this out? Is this functionality wrong?

Reply:
This part of the code is clobbering the submission view. If a data grid has two components x and y, and if x has no data then y's data is getting rendered in the x's column.

Reply:
Okay. Makes sense. I'm a little worried this will cause an issue somewhere else but lets do it.